tests/e2e/precompile-upgrade/suites.go

```diff
@@ -0,0 +1,330 @@
+			}
+			transferAmount := new(big.Int).Div(prevBal, new(big.Int).SetInt64(10))
+
+			ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
+			newBal, err := evmCli.TransferTx(ctx, ewoqAddr, ewoqKey, addr2, transferAmount)
+			cancel()
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
+			err = evmCli.WaitForBalance(ctx, ewoqAddr, newBal)
+			cancel()
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			utils.Outf("{{magenta}}successfully sent:{{/}} after balance %v (old balance %v)\n", newBal, prevBal)
+		})
+
+		// only allow txs from "ewoq"
+		upgradeCfgAllowsEwoq := chainUpgradeConfig{
+			PrecompileUpgrades: []*params.PrecompileUpgrade{
+				{
+					TxAllowListConfig: &precompile.TxAllowListConfig{
+						AllowListConfig: precompile.AllowListConfig{
+							AllowListAdmins: []common.Address{ewoqAddr},
+						},
+						UpgradeableConfig: precompile.UpgradeableConfig{
+							BlockTimestamp: new(big.Int).SetInt64(time.Now().Unix()),
+						},
+					},
+				},
+			},
+		}
+		upgradeCfgAllowsEwoqBytes, err := json.Marshal(upgradeCfgAllowsEwoq)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		ginkgo.By("upgrades all nodes with precompile tx allow list", func() {
+			utils.Outf("{{magenta}}getting cluster status{{/}}\n")
+			ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+			sresp, err := runnerCli.Status(ctx)
+			cancel()
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			for _, name := range sresp.ClusterInfo.NodeNames {
+				upgradeJSONPath := filepath.Join(sresp.ClusterInfo.RootDataDir, name, "chainConfigs", blkChainID, "upgrade.json")
+				gomega.Expect(os.MkdirAll(filepath.Dir(upgradeJSONPath), 0777)).Should(gomega.BeNil())
+
+				utils.Outf("{{magenta}}writing chain config for %q{{/}}: %q\n", name, upgradeJSONPath)
+				gomega.Expect(os.WriteFile(upgradeJSONPath, upgradeCfgAllowsEwoqBytes, 0777)).Should(gomega.BeNil())
+
+				utils.Outf("{{magenta}}restarting the node %q{{/}}\n", name)
+				ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+				resp, err := runnerCli.RestartNode(ctx, name)
+				cancel()
+				gomega.Expect(err).Should(gomega.BeNil())
+
+				time.Sleep(20 * time.Second)
+
+				ctx, cancel = context.WithTimeout(context.Background(), 15*time.Second)
+				_, err = runnerCli.Health(ctx)
+				cancel()
+				gomega.Expect(err).Should(gomega.BeNil())
+				utils.Outf("{{green}}successfully upgraded %q{{/}} (current info: %+v)\n", name, resp.ClusterInfo.NodeInfos)
+			}
+
+			// advance block timestamps by issuing new blocks
+			for i := 0; i < 5; i++ {
+				time.Sleep(5 * time.Second)
+
+				ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+				prevBal, err := evmCli.FetchBalance(ctx, ewoqAddr)
+				cancel()
+				gomega.Expect(err).Should(gomega.BeNil())
+
+				if prevBal.Cmp(new(big.Int).SetInt64(0)) == 0 {
+					ginkgo.Skip("no balance... skipping tests...")
+				}
```

Comment:
This could hide an unexpected failure case, what's the rationale for including this?

Reply:
Removed